### DISCUSSION TITLE
Getting pull request using wrong url?

### DISCUSSION THREAD

**Discussion by TuinboonDev** [octokit/.github#57] [Show and tell]
Hello, im using this code snippet to get a single pull request:

`await context.octokit.rest.pulls.get({
      owner,
      repo,
      number
    })`
    
Except it uses this url: `https://api.github.com/repos/MYGITHUBNAME/REPO/pulls/?number=9` which gives a 404.
While I think it should be: `https://api.github.com/repos/MYGITHUBNAME/REPO/pulls/9` 

Can anyone help me out?

**Comment by gr2m** [comment]
The argument is `pull_number`, not `number`.

**Reply by TuinboonDev** [reply]
The argument is the right number tho, number = 9 so that couldn't be the problem right?

**Reply by gr2m** [reply]
it's

```js
await context.octokit.rest.pulls.get({ owner, repo, pull_number })
```

not

```js
await context.octokit.rest.pulls.get({ owner, repo, number })
```

**Reply by TuinboonDev** [reply]
oh I put it wrong, the "number" is a variable, like:

`const { data: pullRequest } = await context.octokit.rest.pulls.get({
      owner: owner, 
      repo: repo, 
      pull_number: number 
    })`

**Reply by TuinboonDev** [reply]
btw when I use that ^ it gives me a 500

**Reply by gr2m** [reply]
That is not good, it's also not an Octokit problem, the API only responds with 500 if there is an unforeseen problem. I'd contact GitHub support if the problem persists. Try to reproduce the problem with something like `curl` first

**Reply by TuinboonDev** [reply]
Ill try it out, thank you

**Reply by TuinboonDev** [reply]
with curl it works:

`curl -k https://api.github.com/repos/MYGITHUBNAME/REPO/pulls/9`

**Reply by TuinboonDev** [reply]
Also, if I add the argument names like "owner: ..." it gives a 500 and if I remove them its a 404

**Reply by gr2m** [reply]
Can you share a script that reliably reproduces the error? Let me know the exact version of octokit you are using, the version of Node, and what type of authentication you use. Otherwise I won't be able to help further I'm afraid, I'm not sure what's wrong.
